lib/tree.rb

```diff
@@ -1,64 +1,162 @@
 class TreeNode
   attr_reader :key, :value
   attr_accessor :left, :right
-
-   def initialize(key, val)
+  
+  def initialize(key, val)
     @key = key
     @value = val
     @left = nil
     @right = nil
-   end
+  end
 end
 
 class Tree
   attr_reader :root
   def initialize
     @root = nil
   end
-
-  # Time Complexity: 
-  # Space Complexity: 
+  
+  # Time Complexity: Average O(log n), worst-case O(n)
+  # Space Complexity: 1
   def add(key, value)
```

Comment:
Since you're doing a recursive solution, you have to account for the call stack in space complexity.

The space/time complexity is O(log n) if the tree is balanced and O(n) if it's not.